### PR TITLE
fix(SUP-36392):Kaltura Analytics With GTM

### PR DESCRIPTION
**Pull request by inbalvasserman**
### Description of the Changes

issue
gtm missing information about the specific entry send the request

solution
add entry id and entry name to data-layer payload

### CODE REVIEW

File: src/google-tag-manager.ts [hunk 1]
```diff
@@ -78,13 +78,21 @@ export class GoogleTagManager extends BasePlugin<GoogleTagManagerConfig> {
...
     if (!this.allPlayerEvents.has(customEvent)) {
       this.logger.warn(`'${customEvent}' is an invalid player event`);
       return;
     }
+
     this.eventManager.listen(this.player, customEvent, (event: FakeEvent) => {
-      const dataLayerVariablePayload = event.payload !== undefined ? { [`${event.type}-payload`]: event.payload } : {};
+      const eventPayload = event.payload !== undefined ? event.payload : {};
+      //@ts-ignore
+      eventPayload['entry_name'] = this.player.sources.metadata.name;
```
Comment (JonathanTGold) [inline src/google-tag-manager.ts:91]: payload could not be primitive value ?

**Review by JonathanTGold — CHANGES_REQUESTED** [review]

File: src/google-tag-manager.ts (hunk 1 above)
Comment (JonathanTGold) [reply src/google-tag-manager.ts:90]: Inbal: the payload is never 'primitive' it is either `object` or `undefined`

File: src/google-tag-manager.ts [hunk 2]
```diff
@@ -83,8 +83,15 @@ export class GoogleTagManager extends BasePlugin<GoogleTagManagerConfig> {
...
```
Comment (JonathanTGold) [inline src/google-tag-manager.ts:90]:
```suggestion
 if(typeof this.player.sources.metadata.entryId === 'string') {
       //@ts-ignore
      eventPayload['entry_name'] = this.player.sources.metadata.name;
      //@ts-ignore
      eventPayload['entry_id'] = this.player.sources.metadata.entryId;
   }
```

**Review by JonathanTGold — CHANGES_REQUESTED** [review]

**Review by JonathanTGold — APPROVED** [review]

**Review by JonathanTGold — CHANGES_REQUESTED** [review]
the https://github.com/kaltura/kaltura-player-js/pull/616 PR should be approved merged 
and than the kaltura-player-js dependency should be updated in the yarn lock here
and only then the ts build will pass

File: src/google-tag-manager.ts [hunk 3]
```diff
@@ -83,8 +83,15 @@ export class GoogleTagManager extends BasePlugin<GoogleTagManagerConfig> {
       this.logger.warn(`'${customEvent}' is an invalid player event`);
       return;
     }
+
     this.eventManager.listen(this.player, customEvent, (event: FakeEvent) => {
-      const dataLayerVariablePayload = event.payload !== undefined ? { [`${event.type}-payload`]: event.payload } : {};
+      const eventPayload = event.payload !== undefined ? event.payload : {};
```
Comment (SivanA-Kaltura) [inline src/google-tag-manager.ts:88]: you can do event.payload ? ...
unless you specifically want to test for undefined as opposed to null

File: src/google-tag-manager.ts [hunk 4]
```diff
@@ -83,8 +83,15 @@ export class GoogleTagManager extends BasePlugin<GoogleTagManagerConfig> {
...
+      if (typeof this.player.sources.metadata.entryId === 'string') {
```
Comment (SivanA-Kaltura) [inline src/google-tag-manager.ts:89]: why does it matter if entryId is set ? and why does it guarantee that name will be set as well ?

**Review by SivanA-Kaltura — CHANGES_REQUESTED** [review]

File: src/google-tag-manager.ts (hunk 3 above)
Comment (inbalvasserman) [reply src/google-tag-manager.ts:88]: event.payload could be 0 for example, compare to undefine is safer according to yoni.

**Review by JonathanTGold — APPROVED** [review]

**Review by SivanA-Kaltura — APPROVED** [review]

**Pull request merged**